source/Framework Basics.rst

```diff
@@ -0,0 +1,173 @@
+Framework Basics
+=================
+
+Interventions
+-------------
+Interventions can help you execute specific actions based on the current state of the simulation. The simulation engine of Bharat-Sim checks if an inactive intervention can be activated based on the given conditions during every tick. All the active interventions will be recorded in the context and the user can query the context using ``context.activeInterventionNames.contains(interventionName)``.
+
+There are 4 different kinds of Interventions available in Bharat-Sim:
+
+1.Intervention
+~~~~~~~~~~~~~~~~~~~~~
+This is a Generic intervention that can be invoked several times or once when the activation condition is met based on the parameters passed.
+This can be used to implement lockdows during an epidemic everytime the infected fraction of agent reaches a certain threshold.
+There are 5 parameters that can be passed ``Intervention`` object:
+
+    * ``interventionName``: This is the unique name of the intervention.
```

Comment:
Please specify which parameters are compulsory, and which are optional. Do this for the rest of the document as well.